lib/dynamoid/adapter_plugin/aws_sdk_v2.rb

```diff
@@ -463,7 +494,7 @@ def expected_stanza(conditions = nil)
         expected = Hash.new { |h,k| h[k] = {} }
         return expected unless conditions
 
-        conditions[:unless_exists].try(:each) do |col|
+        conditions.delete(:unless_exists).try(:each) do |col|
```

Comment:
Unsure if some reason why not at a glance, but should the `condiitions[:if]` below also be a `delete` for symmetry/POLA purposes perhaps?